app/store/localization-context.tsx

```diff
@@ -1,11 +1,36 @@
 import useMainQuery from "@app/hooks/use-main-query"
 import { useI18nContext } from "@app/i18n/i18n-react"
 import { getLanguageFromLocale } from "@app/utils/locale-detector"
-import { useEffect } from "react"
+import React, { createContext, useCallback, useEffect, useState } from "react"
+
+type LocalizationContextType = {
+  displayCurrency: string
+  setDisplayCurrency: React.Dispatch<React.SetStateAction<string>>
+  convertUsdToDisplayCurrency: (usdAmount: number) => string
+}
+
+export const LocalizationContext = createContext<LocalizationContextType>(null)
 
 export const LocalizationContextProvider = ({ children }) => {
   const { userPreferredLanguage } = useMainQuery()
   const { locale, setLocale } = useI18nContext()
+  const [displayCurrency, setDisplayCurrency] = useState("USD")
+
+  const convertUsdToDisplayCurrency = useCallback(
+    (usdAmount: number) => {
```

Comment:
Can we use `UsdPaymentAmount` here for the type? In general, I think we want to be using the types in this [file](app/types/amounts.ts) because as we start to work with multiple currencies, it will be really important for us to know what currencies we are dealing with.

Reply:
I've moved the function into the custom hook as I think that is a better pattern to follow.  Can discuss in front end sync.